src/generics/utils.js

```diff
@@ -552,6 +591,122 @@ function validateFilters(input, validationData, modelName) {
 	return input
 }
 
+const algorithm = process.env.EMAIL_ID_ENCRYPTION_ALGORITHM
+const secretKey = Buffer.from(process.env.EMAIL_ID_ENCRYPTION_KEY, 'hex')
+const fixedIV = Buffer.from(process.env.EMAIL_ID_ENCRYPTION_IV, 'hex')
+
+const encrypt = (plainTextEmail) => {
+	try {
+		const cipher = crypto.createCipheriv(algorithm, secretKey, fixedIV)
+		return cipher.update(plainTextEmail, 'utf-8', 'hex') + cipher.final('hex')
+	} catch (err) {
+		console.log(err)
+		throw err
+	}
+}
+
+const decrypt = (encryptedEmail) => {
+	try {
+		const decipher = crypto.createDecipheriv(algorithm, secretKey, fixedIV)
+		return decipher.update(encryptedEmail, 'hex', 'utf-8') + decipher.final('utf-8')
+	} catch (err) {
+		console.log(err)
+		throw err
+	}
+}
+
+function extractFilename(fileString) {
+	const match = fileString.match(/([^/]+)(?=\.\w+$)/)
+	return match ? match[0] : null
+}
+
+const generateRedisConfigForQueue = () => {
+	const parseURL = new URL(process.env.REDIS_HOST)
+	return {
+		connection: {
+			host: parseURL.hostname,
+			port: parseURL.port,
+		},
+	}
+}
+
+const generateFileName = (name, extension) => {
+	const currentDate = new Date()
+	const fileExtensionWithTime = moment(currentDate).tz('Asia/Kolkata').format('YYYY_MM_DD_HH_mm') + extension
+	return name + fileExtensionWithTime
+}
+
+function generateCSVContent(data) {
+	if (data.length === 0) {
+		return 'No Data Found'
+	}
+
+	const headers = Object.keys(data[0])
+
+	const csvRows = data.map((row) => {
+		return headers
+			.map((fieldName) => {
+				const value = row[fieldName]
+				if (typeof value === 'object' && value !== null && !Array.isArray(value)) {
+					// Stringify object values and enclose them in double quotes
+					return '"' + JSON.stringify(value) + '"'
+				} else if (Array.isArray(value)) {
+					// Join array values with comma and space, and enclose them in double quotes
+					return '"' + value.join(', ') + '"'
+				} else {
+					return JSON.stringify(value)
+				}
+			})
+			.join(',')
+	})
+	return [headers.join(','), ...csvRows].join('\n')
+}
+
+const addDurationToTime = (timeString, durationMinutes) => {
+	// Parse the time string
+	const trimmedTimeString = timeString.replace(' Hrs', '')
+	const [hoursStr, minutesStr] = trimmedTimeString.split(':')
+	let hours = parseInt(hoursStr, 10)
+	let minutes = parseInt(minutesStr, 10)
+	// Add the duration to the hours and minutes
+	hours += Math.floor(durationMinutes / 60)
+	minutes += durationMinutes % 60 // Remainder gives updated minutes
+	// Format the updated time
+	const updatedTimeString = `${hours.toString().padStart(2, '0')}:${minutes.toString().padStart(2, '0')}`
+	return updatedTimeString
+}
+
+function convertToEpochTime(dateString, timeString, timeZone) {
```

Comment:
Any reason to have a completly new function for this?
Cant this be handled by moment itself?

Reply:
@nevil-mathew so we have Time timeZone and and duration separatly in csv , in portal front end will convert the time to epoach and will send it to backend , so here its only backend and we need to convert to epoach time , so we are using this .

Reply:
Sure i got that, but what i was asking is the need for separate function?
Conversion can be easily handled moment().unix() right?

Reply:
Sure got that, but what I was asking is the need for separate function?
Conversion can be easily handled moment().unix() right?

Reply:
@nevil-mathew will have to look into it , and will change if it is possible